### ISSUE TITLE
Iryna: Incremental -development, week 1, homework

### ISSUE THREAD

**Issue by IrynaSpyrydonova** [HackYourFutureBelgium/class-9-10#158]
# Week 1

## Incremental Development

- [assignments](https://github.com/hackyourfuturebelgium/module-name/tree/master/week-n)

## Prep Work

- [x] Git
- [x] Agile Development
- [x] Relative vs. Absolute Paths
- [x] Inspecting the DOM
- [x] IHTML 5 Semantic Layout
---

## In Class

- [x] Isolate: [live](https://irynaspyrydonova.github.io/Zoom-Group/), [source](https://github.com/IrynaSpyrydonova/Zoom-Group)
- [x] Integrate: [live](https://irynaspyrydonova.github.io/Development-Strategy/), [source](https://github.com/IrynaSpyrydonova/Development-Strategy)

---

### Suggested Study

- [x] Git & GitHub
- [x] HTML & CSS

#### Exercises:

- [x] :egg: html-css-git-exercises - Week 1: [repo](https://github.com/IrynaSpyrydonova/html-css-git-exercises/tree/master/week-1)
- [x] :hatching_chick:HTML-CSS-Practice-Problems
- [ ] :hatched_chick: css-exercises 

### Project

acme-web-design:

- [x] [repo](https://github.com/IrynaSpyrydonova/acme-web-design/)
- [x] [live](https://irynaspyrydonova.github.io/acme-web-design/)
- [x] [development-strategy](https://github.com/IrynaSpyrydonova/acme-web-design/blob/master/development-strategy.md)
- [x] One branch per step in my `development-strategy.md`
- [x] A complete [README ](https://github.com/IrynaSpyrydonova/acme-web-design/blob/master/README.md)

**Comment by Mert1980** [comment]
Well done @IrynaSpyrydonova ! 
I add a few comments below
- Consider adding a lang attribute to the html start tag to declare the language of this document. From line 1, column 16; to line 2, column 6
- Consider using the h1 element as a top-level heading only (all h1 elements are treated as top-level headings by many screen readers and other tools). From line 37, column 9; to line 37, column 12
- Consider using the h1 element as a top-level heading only (all h1 elements are treated as top-level headings by many screen readers and other tools). from line 30, column 9; to line 30, column 12.
- For a good commit messages, you can use the following guidelines in your upcoming projects:

Message Structure
A commit messages consist of three distinct parts separated by a blank line: the title, an optional body, and an optional footer. The layout looks like this:

type: subject
body
footer
The title consists of the type of the message and subject.

The Type
The type is contained within the title and can be one of these types:

feat: a new feature
fix: a bug fix
docs: changes to documentation
style: formatting, missing semicolons, etc; no code change
refactor: refactoring production code
test: adding tests, refactoring test; no production code change
chore: updating build tasks, package manager configs, etc; no production code change

The Subject
Subjects should be no greater than 50 characters, should begin with a capital letter and do not end with a period.

Use an imperative tone to describe what a commit does, rather than what it did. For example, use change; not change or changes.

The Body
Not all commits are complex enough to warrant a body, therefore it is optional and only used when a commit requires a bit of explanation and context. Use the body to explain the what and why of a commit, not the how.

When writing a body, the blank line between the title and the body is required and you should limit the length of each line to no more than 72 characters.

The Footer
The footer is optional and is used to reference issue tracker IDs.

Example Commit Message
feat: Summarize changes in around 50 characters or less

More detailed explanatory text, if necessary. Wrap it to about 72
characters or so. In some contexts, the first line is treated as the
subject of the commit and the rest of the text as the body. The
blank line separating the summary from the body is critical (unless
you omit the body entirely); various tools like `log`, `shortlog`
and `rebase` can get confused if you run the two together.

Explain the problem that this commit is solving. Focus on why you
are making this change as opposed to how (the code explains that).
Are there side effects or other unintuitive consequenses of this
change? Here's the place to explain them.

Further paragraphs come after blank lines.

 - Bullet points are okay, too

 - Typically a hyphen or asterisk is used for the bullet, preceded
   by a single space, with blank lines in between, but conventions
   vary here

If you use an issue tracker, put references to them at the bottom,
like this:

Resolves: #123
See also: #456, #789

**Comment by IrynaSpyrydonova** [comment]
Dear @Mert1980, thank you very much for your detailed response. I really appreciate that and I am sorry in advance for bothering you with additional questions. 
It is not entirely clear for me how I can implement all those changes in order to finish my homework tasks until tomorrow. 
Would it be enough to add a new branch for the adding language tag and amended h1 in my HTML? Or would you suggest an alternative solution?
Thank you in advance.

**Comment by Mert1980** [comment]
You're welcome @IrynaSpyrydonova , coming to your questions; there is no need to create an additional branch in order to add a **language tag**. You just go to the previous branch (**git checkout <branch name>**) that you solved the first exercise and make the necessary change. Then `**git add .**` and `**git commit -m "your commit message"**`. Afterward you push it to git hub `**git push origin <your branch name>**`. Then in your GitHub repo, you **merge your updated branch into the master branch.** 
For the previous commit messages, there is nothing to do, but for the following ones; using an **imperative language** and add **pre-info** like (feat, fix, docs, style, refactor, test, chore) is kind of convention.

**Comment by snicoll** [comment]
@IrynaSpyrydonova I am flagging this as done based on the follow up of the initial feedback. Good job. 

A couple more comments from my side:

* You've used PRs to merge your branches rather than merging them on your computer. If you get a chance, please exercise this as it was part of the objectives for week 1.
* Your development strategy doesn't really have a user story title. If you compare the ones that were provided to you in other exercises you've seen things like "As a user, I ..". These are important as they describe the added value. What feature do you bring to the site? Your structure is therefore a bit too technical and not enough features-oriented. If you have time, look at [this development strategy for the same exercise](https://github.com/adekimpianna/acme-web-design/blob/master/development-strategy.md).

**Comment by IrynaSpyrydonova** [comment]
@snicoll thank you for very much for your time and for your review. 


But few things are still not clear for me. I didn't understand what do you mean with "flagging this as done based on the follow up of the initial feedback." Do you mean, that for your opinion it's not done, but because your comments were not mentioned before, so you flagged that as done? 


I was about to finish the project for week 2, but now I am blocked.

I still don't understand how I suppose to write commit messages. Usually I do the following way - 
`git add .` then `git commit -m "Added the header  and styled it"`.  Thanks to your Sunday classwork, I am now aware that I can use also `git commit` and in the `vim` to put something more. Could some of you please show me, how are you expecting to see my commit messages, and is it wrong to use my way with `git commit -m ""` ?


Regarding the PR. From the beginning I followed the next steps in order to merge the PR from my terminal: `git add .`, ` git commit -m ""`, `git push origin <name of branch>`, `git checkout master`, `git merge <name of the branch>`, `git push origin master`. But in the end, my network looked like a straight line with dots, but according to our discussions I assumed that you prefer to see the following picture 
<img width="285" alt="Screenshot 2020-05-28 at 10 41 56" src="https://user-images.githubusercontent.com/61011137/83119398-f35f3980-a0cf-11ea-8819-85d94a9acb51.png">. 

So could you please correct my commands, that I would be sure that I am doing right.


Talking about dev-strategy, my bad about titles, but I am not entirely sure what do you mean with '' not enough features-oriented"?  I think I have a bit different meaning of the features in this project 😃. For instance, my logic was to apply the header and nav links for the web site. So I created branch, called it `header`, and on this branch, I wrote the code for the header and nav links. 

I asked all these questions only because I want to understand either I do my current homework right or I will make the same mistakes again.

Thanks in advance.

**Comment by snicoll** [comment]
>  I didn't understand what do you mean with "flagging this as done based on the follow up of the initial feedback." Do you mean, that for your opinion it's not done, but because your comments were not mentioned before, so you flagged that as done?

I mean that week 1 is over now and we can always discuss on small details here and there but I felt that it was good enough based on the follow-up on the first review you had.

> I was about to finish the project for week 2, but now I am blocked.

Please don't ask questions about week 2 on another issue, it makes it very hard for us to track who is blocked and who needs support. If you hare blocked on your week 2 assignment, please move your `week-2` issue to `BLOCKED`. Regardless, I am going to answer here this time.

> I still don't understand how I suppose to write commit messages

You got [some very good feedback in this issue](https://github.com/HackYourFutureBelgium/class-9-10/issues/158#issuecomment-632017348), please review them. There is nothing wrong writing your commit message with `git commit -m ""`. However, you've seen in the assignement this week that linking the commit to an issue is important. In order to do that, you have to write a multi-lines commit message such as:

```
Add site header

Closes gh-4
```

Writing a commit message that is mutli-lines on the command line is hard so I wouldn't not use `git commit -m`. Find yourself a tool that you like and ask on the main slack if you need support.

> But in the end, my network looked like a straight line with dots, but according to our discussions I assumed that you prefer to see the following picture

To be crystal clear, none of the comments I've made anywhere so far are meant to express a preference but rather to guide you. The only thing we're asking is that the instructions of the exercises are followed. There's nothing there that states whether we want a "fast-forward merge" (your way, no little bumps) or a "merge commit" (little bumps as in your image). What we're asking is that you create a branch for each story and merge it locally before starting the next story. The commands you've added are doing that so all is fine from my perspective.

> I am not entirely sure what do you mean with '' not enough features-oriented"? I think I have a bit different meaning of the features in this project 😃. For instance, my logic was to apply the header and nav links for the web site. So I created branch, called it header, and on this branch, I wrote the code for the header and nav links.

I might have been a bit too fast here as we haven't started agile development just yet. Take the stories like a discussion with a customer. A customer will not ask you to add a footer but it might ask you to make sure there is a copyright information to be compliant with the law. 

Look also to [the template](https://github.com/HackYourFutureBelgium/incremental-development/blob/master/integrate/complete-development-strategy.md) we've been using in the past. Each story express a feature the visitor of the site could use/see. The "too technical" bits is not what's important in my feedback. The fact that it's not enough features oriented is. 

If you have question about week-1 please follow-up here. Otherwise ping me in your week-2 issue or on slack :) Hope that helps.

**Comment by IrynaSpyrydonova** [comment]
Thank you 🙂

I really appreciate the comment about messages, I just didn't get how to apply that in practice. And wasn't sure that I am doing that correctly. So I am grateful to both of you for your patience in the explanations.

 We have already started the Agile, but I were assumed that dev-strategy is only for developers,  to help others to understand which code I was writing on that branch. 
So thank you for your clarification one more time, I will try to write my next dev-strategy more like discussion with a customer.
